x/pairing/keeper/msg_server_stake_provider_test.go

```diff
@@ -847,19 +849,291 @@ func TestCommisionChange(t *testing.T) {
+			afterOperator := ts.GetBalance(tt.operator)
+
+			ts.AdvanceEpoch()
+
+			if !tt.valid {
+				require.Error(t, err)
+
+				// balance
+				require.Equal(t, beforeVault, afterVault)
+				require.Equal(t, beforeOperator, afterOperator)
+
+				// stake entry
+				_, found := ts.Keepers.Epochstorage.GetStakeEntryByAddressCurrent(ts.Ctx, tt.spec.Index, tt.operator.String())
+				require.True(t, found) // should be found because operator is registered in a vaild stake entry
+
+				// delegations
+				res, err := ts.QueryDualstakingDelegatorProviders(tt.vault.String(), false)
+				require.NoError(t, err)
+				require.Len(t, res.Delegations, 0)
+			} else {
+				require.NoError(t, err)
+
+				// balance
+				require.Equal(t, beforeVault-testStake, afterVault)
+				if !tt.operator.Equals(tt.vault) {
+					require.Equal(t, beforeOperator, afterOperator)
+				} else {
+					require.Equal(t, beforeOperator-testStake, afterOperator)
+				}
+
+				// stake entry
+				stakeEntry, found := ts.Keepers.Epochstorage.GetStakeEntryByAddressCurrent(ts.Ctx, tt.spec.Index, tt.operator.String())
+				require.True(t, found)
+				require.Equal(t, testStake, stakeEntry.Stake.Amount.Int64())
+				require.Equal(t, int64(0), stakeEntry.DelegateTotal.Amount.Int64())
+
+				// delegations
+				res, err := ts.QueryDualstakingDelegatorProviders(tt.vault.String(), false)
+				require.NoError(t, err)
+				require.Len(t, res.Delegations, 1)
+				require.Equal(t, tt.operator.String(), res.Delegations[0].Provider)
+				require.Equal(t, testStake, res.Delegations[0].Amount.Amount.Int64())
+			}
+		})
+	}
+}
+
+// TestVaultOperatorExistingStakeEntry tests that staking (or "self delegation") to an existing stake entry
+// only works when the creator is the vault and not the operator.
+// For each scenario, we'll check the vault's balance, the stake entry's stake/delegate total fields and the delegations
+// fixation store.
+// Scenarios:
+// 1. stake with vault to an existing stake entry -> should work
+// 2. try with operator -> should fail
+func TestVaultOperatorExistingStakeEntry(t *testing.T) {
+	ts := newTester(t)
+
+	p1Acc, _ := ts.AddAccount(common.PROVIDER, 0, testBalance)
+	err := ts.StakeProviderExtra(p1Acc.Addr.String(), p1Acc.Vault.Addr.String(), ts.spec, testStake, []epochstoragetypes.Endpoint{{Geolocation: 1}}, 1, "test")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		vault    sdk.AccAddress
+		operator sdk.AccAddress
+		spec     spectypes.Spec
+		valid    bool
+	}{
+		{"stake with vault", p1Acc.Vault.Addr, p1Acc.Addr, ts.spec, true},
+		{"stake with operator", p1Acc.Addr, p1Acc.Addr, ts.spec, false},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			beforeVault := ts.GetBalance(tt.vault)
+			beforeOperator := ts.GetBalance(tt.operator)
+			err := ts.StakeProviderExtra(tt.operator.String(), tt.vault.String(), tt.spec, testStake+(100*int64(i+1)), []epochstoragetypes.Endpoint{{Geolocation: 1}}, 1, "test")
```

Comment:
Can you please export the calculation there to a new line with an explanatory name?